src/flb_engine_dispatch.c

```diff
@@ -75,7 +75,15 @@ int flb_engine_dispatch_retry(struct flb_task_retry *retry,
     flb_event_chunk_update(task->event_chunk, buf_data, buf_size);
 
     /* flush the task */
-    ret = flb_output_task_flush(task, retry->o_ins, config);
```

Comment:
The ret value was used here in the retry code... Need to handle retry errors properly.